### PR TITLE
Modify mention user and mention report extras field names in ExpensiMark htmlToText and htmlToMarkdown rules

### PR DESCRIPTION
**Pull request by tsa321**
This PR changes field name of extras in ExpensiMark to fix type check fails.
Modify mention user and mention report extras field names in ExpensiMark htmlToText and htmlToMarkdown rules

### Fixed Issues
$ https://github.com/Expensify/App/issues/40928

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
typecheck on expensify/App should run successfully

1. What tests did you perform that validates your changed worked?
I have tested it on my other PR, thread report from report action that contains mention room and mention user should display report name properly in header report and in LHN.

# QA
1. What does QA need to do to validate your changes?
N/A

1. What areas to they need to test for regressions?
N/A

### CODE REVIEW

**Comment by tsa321** [comment]
I have read the CLA Document and I hereby sign the CLA

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -547,15 +547,18 @@ export default class ExpensiMark {
...
+                        const accountToNameMap = extras.accountIDToName;
+                        if (!accountToNameMap || !accountToNameMap[g1]) {
+                            Log.alert('[ExpensiMark] Missing account name', {accountID: g1});
+                            return '@Hidden';
+                        }
 
-                    return `@${extras.accountIdToName[g1]}`;
+                        return `@${accountToNameMap[g1]}`;
+                    }
+                    return Str.removeSMSDomain(g2)
```
Comment (mollfpr) [inline lib/ExpensiMark.js:561]: What's the purpose of these changes?

**Review by mollfpr — COMMENTED** [review]
It seems we have a test failure.

Screenshot 2024-05-28 at 17 32 06

I'll be expecting the changes to minimal as it could so we don't break anything here.

File: lib/ExpensiMark.js (hunk 1 above)
Comment (luacmartins) [reply lib/ExpensiMark.js:561]: +1

Comment (tsa321) [reply lib/ExpensiMark.js]: The failling tests are because I didn't changes the fields in the test files yet, I am sorry.

For  `accountToNameMap`, I am following the pattern in the file for `reportToNameMap`:

https://github.com/Expensify/expensify-common/blob/321e4a7f5e618000be812d5b7dc76d00f81084f9/lib/ExpensiMark.js#L545

`accountToNameMap` is just a variable that store `extras.accountIdToName`. It is the same thing.
Should I revert it?

**Comment by tsa321** [comment]
@mollfpr The failling tests are because I didn't changes the fields in the test files yet, I am sorry. Should be fine now...

**Review by luacmartins — APPROVED** [review]
